File: src/main/java/httpServer/httpLogic/Response.java
package httpServer.httpLogic;

import java.util.Map;

public class Response {
    protected String httpVersion;
    protected String statusCode;
    protected String statusMessage;
    protected Map<String, String> headers;
    protected String body;

    public Response(String httpVersion,
                    String statusCode,
                    String statusMessage,
                    Map<String, String> headers,
                    String body) {
        this.httpVersion = httpVersion;
        this.statusCode = statusCode;
        this.statusMessage = statusMessage;
        this.headers = headers;
        this.body = body;
    }

}

File: http_server_spec/features/support/response.rb
class Response

  def initialize(httpartyResponse)
    self.httpartyResponse = httpartyResponse
  end

  def location
    httpartyResponse.headers["location"]
  end

  def allowed_headers
    httpartyResponse.headers["allow"].split(/[ \t]*,[ \t]*/)
  end

  def body
    httpartyResponse.body.to_s
  end

  def status_code
    httpartyResponse.code
  end

  def self.get_status_code_from_raw_string_response(response)
    response.split(" ")[1].to_i
  end

  private

  attr_accessor :httpartyResponse


end
